src/firefly/js/ui/SimpleComponent.jsx

```diff
@@ -60,12 +60,14 @@ export function useStoreConnector(stateGetter, deps=[]) {
 
     let isMounted = true;
     useEffect(() => {
+        let cState = val;
```

Comment:
Well I think i understand why this line fixes it but it is tricky. This is a double closure with `val` is not really the old value.

Reply:
I didn't get it, I went to Loi's office with ??? "double closure", it is!